packages/flutter/lib/src/rendering/object.dart

```diff
@@ -57,190 +59,215 @@ class PaintingCanvas extends Canvas {
 /// not hold a reference to the canvas across operations that might paint
 /// child render objects.
 class PaintingContext {
-  /// Construct a painting context at a given offset with the given bounds
-  PaintingContext.withOffset(Offset offset, Rect paintBounds) {
-    _containerLayer = new ContainerLayer(offset: offset);
-    _startRecording(paintBounds);
+  PaintingContext._(this._containerLayer, this._paintBounds);
+
+  final ContainerLayer _containerLayer;
+  final Rect _paintBounds;
+
+  /// Repaint the given render object.
+  ///
+  /// The render object must have a composited layer and must be in need of
+  /// painting. The render object's layer is re-used, along with any layers in
+  /// the subtree that don't need to be repainted.
+  static void repaint(RenderObject root) {
```

Comment:
Why have this method, rather than having RenderObject call _repaintCompositedChild() directly?